MvvmHelpers/WeakEventManager.cs

```diff
@@ -0,0 +1,148 @@
+			where TEventArgs : EventArgs
+		{
+			if (IsNullOrEmpty(eventName))
+				throw new ArgumentNullException(nameof(eventName));
+
+			if (handler == null)
+				throw new ArgumentNullException(nameof(handler));
+
+			AddEventHandler(eventName, handler.Target, handler.GetMethodInfo());
+		}
+
+		public void AddEventHandler(EventHandler handler, [CallerMemberName]string eventName = null)
+		{
+			if (IsNullOrEmpty(eventName))
+				throw new ArgumentNullException(nameof(eventName));
+
+			if (handler == null)
+				throw new ArgumentNullException(nameof(handler));
+
+			AddEventHandler(eventName, handler.Target, handler.GetMethodInfo());
+		}
+
+		public void HandleEvent(object sender, object args, string eventName)
+		{
+			var toRaise = new List<(object subscriber, MethodInfo handler)>();
+			var toRemove = new List<Subscription>();
+
+			if (eventHandlers.TryGetValue(eventName, out var target))
+			{
+				for (var i = 0; i < target.Count; i++)
+				{
+					var subscription = target[i];
+					var isStatic = subscription.Subscriber == null;
+					if (isStatic)
+					{
+						// For a static method, we'll just pass null as the first parameter of MethodInfo.Invoke
+						toRaise.Add((null, subscription.Handler));
+						continue;
+					}
+
+					var subscriber = subscription.Subscriber.Target;
+
+					if (subscriber == null)
+						// The subscriber was collected, so there's no need to keep this subscription around
+						toRemove.Add(subscription);
+					else
+						toRaise.Add((subscriber, subscription.Handler));
+				}
+
+				for (var i = 0; i < toRemove.Count; i++)
+				{
+					var subscription = toRemove[i];
+					target.Remove(subscription);
+				}
+			}
+
+			for (var i = 0; i < toRaise.Count; i++)
+			{
+				(var subscriber, var handler) = toRaise[i];
+				handler.Invoke(subscriber, new[] { sender, args });
+			}
+		}
+
+		public void RemoveEventHandler<TEventArgs>(EventHandler<TEventArgs> handler, [CallerMemberName]string eventName = null)
+			where TEventArgs : EventArgs
+		{
+			if (IsNullOrEmpty(eventName))
+				throw new ArgumentNullException(nameof(eventName));
+
+			if (handler == null)
+				throw new ArgumentNullException(nameof(handler));
+
+			RemoveEventHandler(eventName, handler.Target, handler.GetMethodInfo());
+		}
+
+		public void RemoveEventHandler(EventHandler handler, [CallerMemberName]string eventName = null)
+		{
+			if (IsNullOrEmpty(eventName))
+				throw new ArgumentNullException(nameof(eventName));
+
+			if (handler == null)
+				throw new ArgumentNullException(nameof(handler));
+
+			RemoveEventHandler(eventName, handler.Target, handler.GetMethodInfo());
+		}
+
+		void AddEventHandler(string eventName, object handlerTarget, MethodInfo methodInfo)
+		{
+			if (!eventHandlers.TryGetValue(eventName, out var targets))
+			{
+				targets = new List<Subscription>();
+				eventHandlers.Add(eventName, targets);
+			}
+
+			if (handlerTarget == null)
```

Comment:
Use `handler is null` to squeeze out some more performance benefits: https://www.gullberg.tk/blog/is-null-versus-null-in-c/

**tl;dr** Pattern Matching is faster than Equality comparison in the .NET IL